File: snippets/fsharp/System/UIntPtr/op_Addition/op_addition1.fs
module op_addition1

open System

// <Snippet1>
let arr = [| 1; 2; 3; 4; 5; 6; 7; 8; 9; 10 |]
let ptr = UIntPtr(uint arr[0])
for i = 0 to arr.Length - 1 do
    let newPtr = ptr + UIntPtr(uint i)
    printfn $"{newPtr}"
// </Snippet1>  

File: snippets/csharp/System/UIntPtr/op_Addition/op_addition1.cs
using System;

public class UIntPtrAdditionExample
{
    public static void Run()
    {
        // <Snippet1>
        int[] arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
        UIntPtr ptr = (UIntPtr)arr[0];
        for (int ctr = 0; ctr < arr.Length; ctr++)
        {
            UIntPtr newPtr = ptr + (nuint)ctr;
            Console.WriteLine(newPtr);
        }
        // </Snippet1>
    }
}
